tests/test_gradient.py

```diff
@@ -51,6 +53,71 @@ def test_gradient_checkpointing(self, shape, kernel, space_order):
         gradient2, _ = wave.jacobian_adjoint(residual, u0, vp=v0, checkpointing=False)
         assert np.allclose(gradient.data, gradient2.data)
 
+    @pytest.mark.parametrize('space_order', [4])
+    @pytest.mark.parametrize('kernel', ['OT2'])
+    @pytest.mark.parametrize('shape', [(70, 80)])
+    def test_gradient_by_parts(self, shape, kernel, space_order):
+        preset = 'layers-isotropic'
+        nbl = 10
+        dtype = np.float32
+        spacing = (10, 10)
+        tn = 500.
+
+        model = demo_model(preset, space_order=space_order, shape=shape, nbl=nbl,
+                           dtype=dtype, spacing=spacing)
+        m = model.m
+        geometry = setup_geometry(model, tn)
+        dt = geometry.dt
+        u = TimeFunction(name='u', grid=model.grid, time_order=2, space_order=space_order,
+                         save=geometry.nt)
+
+        src = geometry.src
+
+        rec = geometry.rec
+
+        s = model.grid.stepping_dim.spacing
+
+        eqn = iso_stencil(u, model, kernel)
+
+        # Construct expression to inject source values
+        src_term = src.inject(field=u.forward, expr=src * s**2 / m)
+
+        # Create interpolation expression for receivers
+        rec_term = rec.interpolate(expr=u)
+
+        fwd_op = Operator(eqn + src_term + rec_term, subs=model.spacing_map,
+                          name='Forward')
```

Comment:
Can probaly get all of that with something like

```
_, _, _, u, rec = acoustic.run(...., sav=true)
```

Reply:
Since I have to setup the gradient operator by hand anyway, I thought it makes sense to keep it consistent between the forward and the gradient?

Reply:
I get you, but since we only care about the gradient for a given `u` , makes the test bit less verbose. But if you prefer like that it's fine.